src/main/scala/SubscriptionActor.scala

```diff
@@ -0,0 +1,84 @@
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import akka.util.Timeout
+import generic.Event
+import sangria.ast.OperationType
+import sangria.execution.{Executor, PreparedQuery}
+import sangria.marshalling.sprayJson._
+import sangria.parser.QueryParser
+import spray.json._
+
+import scala.util.{Failure, Success}
+
+object SubscriptionActor extends DefaultJsonProtocol {
+  sealed trait SubscriptionMessage
+  case class Subscribe(query: String, operation: Option[String])
+  case class SubscriptionAccepted() extends SubscriptionMessage
+  case class QueryResult(json: JsValue) extends SubscriptionMessage
+  case class Connected(outgoing: ActorRef)
+  case class PreparedQueryContext(query: PreparedQuery[Ctx, Any, JsObject])
+
+  implicit val subscribeProtocol = jsonFormat2(Subscribe)
+}
+
+class SubscriptionActor(publisher: ActorRef, ctx: Ctx)(implicit timeout: Timeout) extends Actor with ActorLogging {
+
+  import SubscriptionActor._
+
+  implicit val ec = context.system.dispatcher
+
+  val executor = Executor(schema.createSchema)
+  var subscriptions = Map.empty[String, Set[PreparedQueryContext]]
+
+  override def receive: Receive = {
+    case Connected(outgoing) =>
+      publisher ! SubscriptionEventPublisher.Join
+      context.become(connected(outgoing))
+  }
+
+  def connected(outgoing: ActorRef): Receive = {
+    case e: Subscribe =>
+      log.info(s"Got sub: $e")
+      prepareQuery(e)
+
+    case context: PreparedQueryContext =>
+      log.info(s"Query is prepared: $context")
+      outgoing ! SubscriptionAccepted()
+      context.query.fields.map(_.field.name).foreach { field =>
+        subscriptions = subscriptions.updated(field, subscriptions.get(field) match {
+          case Some(contexts) => contexts + context
+          case _ => Set(context)
+        })
+      }
+
+    case event: Event =>
+      val fieldName = schema.subscriptionFieldName(event)
+      queryContextsFor(fieldName) foreach { ctx =>
+        ctx.query.execute(root = event) map { result =>
+          outgoing ! QueryResult(result)
+        }
```

Comment:
It's the same issue with error handling here (as in the `executor.prepare` code below). I understand that this code is just an example/demonstration of the concept, so some of the details are skipped, Still I think error handling is important for these 2 operations and for the demonstration itself. Example code is very popular source of copy/paste, so if people copy/past then it's better be with error handling :)